src/client/dfs/dfs.c

```diff
@@ -1688,6 +1688,65 @@ dfs_open(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	return rc;
 }
 
+int
+dfs_dup(dfs_t *dfs, dfs_obj_t *obj, dfs_obj_t **_new_obj)
+{
+	dfs_obj_t	*new_obj;
+	int		rc;
+
+	if (dfs == NULL || !dfs->mounted)
+		return -DER_INVAL;
+	if (obj == NULL || _new_obj == NULL)
+		return -DER_INVAL;
+
+	D_ALLOC_PTR(new_obj);
+	if (new_obj == NULL)
+		return -DER_NOMEM;
+
+	switch (obj->mode & S_IFMT) {
+	case S_IFDIR:
+		rc = daos_obj_open(dfs->coh, obj->oid, DAOS_OO_RW,
+				   &new_obj->oh, NULL);
+		if (rc)
+			D_GOTO(err, rc);
+		break;
+	case S_IFREG:
+	{
+		char		buf[1024];
+		daos_iov_t	ghdl;
+
+		daos_iov_set(&ghdl, buf, 1024);
+
+		rc = daos_array_local2global(obj->oh, &ghdl);
+		if (rc)
+			D_GOTO(err, rc);
+
+		rc = daos_array_global2local(dfs->coh, ghdl, &new_obj->oh);
+		if (rc)
+			D_GOTO(err, rc);
+		break;
+	}
+	case S_IFLNK:
+		new_obj->value = strdup(obj->value);
```

Comment:
Presumably this can't use strndup because value doesn't have a max length?  Do we check that anywhere else?

This is going to fail the memory allocation checking (when we enable it) however, as it should be using D_STRNDUP().

Reply:
i need to have a better fix for all symlink values. this is an issue not just in this place of the code.
so let's punt this for a different patch.